### DISCUSSION TITLE
Spatial and temporal attention.

### DISCUSSION THREAD

**Discussion by llvm-x86** [timeseriesAI/tsai#732] [General]
Do the transformers find relationships between features and time?

Ie: 
`(_, 0, 13)` finds a relationship with `(_, 5, 9)`?

**Comment by oguiza** [comment]
That depends on the Transformer. 
Some of the models in the tsai library (like TST or TSiT) use attention between time steps (all features together). Others (like PatchTST) find relationships between patches of time steps for each feature. 
None of the models currently available uses both spatial and temporal attention.

**Reply by llvm-x86** [reply]
How would I make one?

**Reply by oguiza** [reply]
Hi @shusei-kagari,
This is currently out of the scope of tsai. This would require a considerable amount of changes. You'd need to build a dataset, dataloader, and select an architecture. I'd probably start reviewing the literature to identify an architecture and see if they have a github repo where it's implemented.
